.github/workflows/ci.yml

```diff
@@ -89,43 +89,33 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04, ubuntu-22.04]
-        containerd: [v1.6.21, v1.7.1]
+        containerd: [1.6.21, 1.7.1]
 
     steps:
       - name: Checkout extensions
         uses: actions/checkout@v4
 
+      - name: Download containerd archive
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release download v${{ matrix.containerd }} \
+            --repo containerd/containerd \
+            --pattern 'containerd-${{ matrix.containerd }}-linux-amd64.tar.gz' \
+            --output containerd.tar.gz
+
+      - name: Extract containerd binaries to $HOME/.local/bin
+        run: |
+          mkdir -p $HOME/.local/bin
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+          tar -xf containerd.tar.gz -C $HOME/.local
+
       - name: Checkout containerd
         uses: actions/checkout@v4
         with:
           repository: containerd/containerd
           path: src/github.com/containerd/containerd
-          ref: ${{ matrix.containerd }}
-
-      - name: Get Go Version
-        run: |
-          go_version=$(awk -F': ' '/GO_VERSION/ {gsub(/["'\'']/, "", $2); print $2; exit}' .github/workflows/release.yml)
-          echo "GO_VERSION=$go_version" >> $GITHUB_ENV
-        working-directory: src/github.com/containerd/containerd
-
-      - uses: actions/setup-go@v4
-        with:
-          go-version: ${{ env.GO_VERSION }}
-
-      - name: Install containerd
-        env:
-          GOFLAGS: -modcacherw
-          CGO_ENABLED: 1
-        run: |
-          # Install containerd dependencies first
-          sudo apt-get install -y gperf
-          sudo -E PATH=$PATH script/setup/install-seccomp
-          sudo -E PATH=$PATH script/setup/install-runc
-          sudo -E PATH=$PATH script/setup/install-cni $(grep containernetworking/plugins go.mod | awk '{print $2}')
```

Comment:
nice, I ended up not doing this since I thought cni and other parts where needed but looks like it isn't!